### ISSUE TITLE
Compiler error when doing unsigned math for storage array index

### ISSUE THREAD

**Issue by fubuloubu** [vyperlang/vyper#2185]
### Version Information

* vyper Version (output of `vyper --version`): 0.2.5+commit.a0c561c
* OS: linux
* Python Version (output of `python --version`): 3.8.5
* Environment (output of `pip freeze`): brownie

### What's your issue about?
Error text:
```python
  File "vyper/cli/vyper_json.py", line 394, in compile_json
    compiler_data, warn_data = compile_from_input_dict(input_dict, exc_handler, root_path)
  File "vyper/cli/vyper_json.py", line 326, in compile_from_input_dict
    return exc_handler(contract_path, exc, "compiler"), {}
  File "vyper/cli/vyper_json.py", line 102, in exc_handler_raises
    raise exception
  File "vyper/cli/vyper_json.py", line 318, in compile_from_input_dict
    data = vyper.compile_codes(
  File "vyper/opcodes.py", line 223, in _wrapper
    return fn(*args, **kwargs)
  File "vyper/compiler/__init__.py", line 110, in compile_codes
    raise exc
  File "vyper/compiler/__init__.py", line 105, in compile_codes
    out[contract_name][output_format] = OUTPUT_FORMATS[output_format](compiler_data)
  File "vyper/compiler/output.py", line 54, in build_abi_output
    gas_estimates = build_gas_estimates(compiler_data.lll_nodes)
  File "vyper/compiler/phases.py", line 102, in lll_nodes
    self._gen_lll()
  File "vyper/compiler/phases.py", line 97, in _gen_lll
    self._lll_nodes, self._lll_runtime = generate_lll_nodes(self.source_code, self.global_ctx)
  File "vyper/compiler/phases.py", line 225, in generate_lll_nodes
    lll_nodes, lll_runtime = parser.parse_tree_to_lll(source_code, global_ctx)
  File "vyper/parser/parser.py", line 204, in parse_tree_to_lll
    o, runtime = parse_other_functions(
  File "vyper/parser/parser.py", line 112, in parse_other_functions
    parse_function(
  File "vyper/parser/function_definitions/parse_function.py", line 55, in parse_function
    o = parse_internal_function(code=code, sig=sig, context=context,)
  File "vyper/parser/function_definitions/parse_internal_function.py", line 252, in parse_internal_function
    + [parse_body(c, context) for c in code.body]
  File "vyper/parser/function_definitions/parse_internal_function.py", line 252, in <listcomp>
    + [parse_body(c, context) for c in code.body]
  File "vyper/parser/stmt.py", line 641, in parse_body
    return parse_stmt(code, context)
  File "vyper/parser/stmt.py", line 635, in parse_stmt
    return Stmt(stmt, context).lll_node
  File "vyper/parser/stmt.py", line 41, in __init__
    self.lll_node = fn()
  File "vyper/parser/stmt.py", line 267, in parse_For
    ["repeat", pos, start, rounds, parse_body(self.stmt.body, self.context)],
  File "vyper/parser/stmt.py", line 645, in parse_body
    lll = parse_stmt(stmt, context)
  File "vyper/parser/stmt.py", line 635, in parse_stmt
    return Stmt(stmt, context).lll_node
  File "vyper/parser/stmt.py", line 41, in __init__
    self.lll_node = fn()
  File "vyper/parser/stmt.py", line 103, in parse_If
    add_on = [parse_body(self.stmt.orelse, self.context)]
  File "vyper/parser/stmt.py", line 645, in parse_body
    lll = parse_stmt(stmt, context)
  File "vyper/parser/stmt.py", line 635, in parse_stmt
    return Stmt(stmt, context).lll_node
  File "vyper/parser/stmt.py", line 41, in __init__
    self.lll_node = fn()
  File "vyper/parser/stmt.py", line 110, in parse_If
    body = ["if", test_expr, parse_body(self.stmt.body, self.context)] + add_on
  File "vyper/parser/stmt.py", line 645, in parse_body
    lll = parse_stmt(stmt, context)
  File "vyper/parser/stmt.py", line 635, in parse_stmt
    return Stmt(stmt, context).lll_node
  File "vyper/parser/stmt.py", line 41, in __init__
    self.lll_node = fn()
  File "vyper/parser/stmt.py", line 94, in parse_Assign
    target = self._get_target(self.stmt.target)
  File "vyper/parser/stmt.py", line 627, in _get_target
    target = Expr.parse_variable_location(target, self.context)
  File "vyper/parser/expr.py", line 978, in parse_variable_location
    o = cls(expr, context).lll_node
  File "vyper/parser/expr.py", line 200, in __init__
    self.lll_node = fn()
  File "vyper/parser/expr.py", line 413, in parse_Subscript
    index = Expr.parse_value_expr(self.expr.slice.value, self.context)
  File "vyper/parser/expr.py", line 973, in parse_value_expr
    return unwrap_location(cls(expr, context).lll_node)
  File "vyper/parser/expr.py", line 202, in __init__
    raise TypeCheckFailure(f"{type(node).__name__} node did not produce LLL")
TypeCheckFailure: BinOp node did not produce LLL
```
The code that failed:
```python
MAXIMUM_STRATEGIES: constant(uint256) = 20
withdrawalQueue: public(address[MAXIMUM_STRATEGIES])

@external
def organizeWithdrawalQueue():
    offset: uint256 = 0
    for idx in range(MAXIMUM_STRATEGIES):
        strategy: address = self.withdrawalQueue[idx]
        if strategy == ZERO_ADDRESS:
            offset += 1  # how many values we need to shift, always `<= idx`
        elif offset > 0:
            self.withdrawalQueue[idx-offset] = strategy
            self.withdrawalQueue[idx] = ZERO_ADDRESS
```

### How can it be fixed?
This line is the problem
```python
self.withdrawalQueue[idx-offset] = strategy
```
Remove the subtraction of `offset` and it compiles

**Comment by iamdefinitelyahuman** [comment]
This will be fixed by #2180

**Comment by fubuloubu** [comment]
NOTE: This problem can also be solved if `offset` is type `int128`
